lib/src/widgets/button_widget.dart

```diff
@@ -0,0 +1,340 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+///
+/// Authors: Tony Chen
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io' show File;
+
+import 'package:flutter/foundation.dart' show kIsWeb;
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'package:file_picker/file_picker.dart';
+import 'package:universal_html/html.dart' as html;
+import 'package:markdown_widgets/src/constants/pkg.dart'
+    show defaultFileName, defaultButtonText;
+
+class ButtonWidget extends StatefulWidget {
+  final String command;
+  final Map<String, dynamic> state;
+
+  ButtonWidget({
+    Key? key,
+    required this.command,
+    required this.state,
+  }) : super(key: key);
+
+  @override
+  _ButtonWidgetState createState() => _ButtonWidgetState();
+}
+
+class _ButtonWidgetState extends State<ButtonWidget> {
+  late final String buttonText;
+  late final int actionType;
+  late final String actionParameter;
+
+  @override
+  void initState() {
+    super.initState();
+    _parseCommand();
+  }
+
+  void _parseCommand() {
+    final buttonExp = RegExp(r'%% Button\((.+)\)');
+    final buttonMatch = buttonExp.firstMatch(widget.command);
+
+    if (buttonMatch != null) {
+      final argsString = buttonMatch.group(1)!;
+      final args = _parseArguments(argsString);
+
+      if (args.isNotEmpty) {
+        buttonText = args.isNotEmpty ? args[0] : defaultButtonText;
+        actionType = args.length > 1 ? int.tryParse(args[1]) ?? 0 : 0;
+        actionParameter = args.length > 2 ? args[2] : defaultFileName;
+      } else {
+        buttonText = defaultButtonText;
+        actionType = 0;
+        actionParameter = defaultFileName;
+      }
+    } else {
+      buttonText = defaultButtonText;
+      actionType = 0;
+      actionParameter = defaultFileName;
+    }
+  }
+
+  /// Collects the user responses from the state map.
+  Map<String, dynamic> _collectData() {
+    // Access the state variables and collect data
```

Comment:
Minor thing but we should end sentences with a full stop https://survivor.togaware.com/gnulinux/flutter-style-comments.html